pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -586,6 +593,101 @@ func proxyConfigSamplingValue(config *meshconfig.ProxyConfig) float64 {
+	if proxyCfg == nil || proxyCfg.Tracing == nil {
+		return
+	}
+	var serviceCluster string
+	if node.XdsNode != nil {
+		serviceCluster = node.XdsNode.Cluster
+	}
+
+	switch tracer := proxyCfg.Tracing.Tracer.(type) {
+	case *meshconfig.Tracing_Datadog_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Datadog.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("datadog")
+			return
+		}
+
+		cfg, _ := datadogConfig(serviceCluster, hostname, clusterName)
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyDatadog,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_Lightstep_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Lightstep.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("lightstep")
+			return
+		}
+
+		cfg, _ := otelLightStepConfig(clusterName, hostname, tracer.Lightstep.GetAccessToken())
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyOpenTelemetry,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_OpenCensusAgent_:
+	case *meshconfig.Tracing_Stackdriver_:
+		// stackdriver/opencensus always need bootstrap tracing?
+	case *meshconfig.Tracing_Zipkin_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Zipkin.Address)
+		if err != nil {
+			log.Warnf("get host cluster failed: %v", err)
+			model.IncLookupClusterFailures("zipkin")
+			return
+		}
+		cfg, _ := zipkinConfig(hostname, clusterName, false)
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyZipkin,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	default:
+		log.Warn("Tracing provider not configured")
+	}
+}
+
+func getHostCluster(push *model.PushContext, address string) (string, string, error) {
+	host, port, err := net.SplitHostPort(address)
+	if err != nil {
+		return "", "", fmt.Errorf("could not parse address %q: %v", address, err)
+	}
+	h := model.ResolveShortnameToFQDN(host, config.Meta{
+		Namespace: push.Mesh.RootNamespace,
+		Domain:    push.Mesh.TrustDomain,
+	})
+	p, err := strconv.Atoi(port)
+	if err != nil {
+		return "", "", fmt.Errorf("could not parse address %q: %v", address, err)
+	}
+	hostname, clusterName, err := clusterLookupFn(push, string(h), p)
+	if err != nil {
+		// support short service names for tracing providers, e.g. "zipkin.istio-system:9411"
+		if len(strings.Split(host, ".")) == 2 {
+			hostname, clusterName, err = clusterLookupFn(push, fmt.Sprintf("%s.svc.%s", host, push.Mesh.TrustDomain), p)
+			if err == nil {
+				return hostname, clusterName, nil
+			}
+		}
+
+		if parts := strings.Split(host, "."); len(parts) == 3 && parts[2] == "svc" {
+			hostname, clusterName, err = clusterLookupFn(push, fmt.Sprintf("%s.%s", host, push.Mesh.TrustDomain), p)
```

Comment:
I am not sure why still need this, since you have converted it to FQDN in L661

Reply:
I have same thought as you did. 
If you take a looks at the implementation of `model.ResolveShortnameToFQDN` you will see that
it will not convert to `name.namespace.svc.cluster.local` when input is `name.namespace`.

https://github.com/istio/istio/blob/f9b1e58a01bed1a6cfb594ffca998aecf2cbca16/pilot/pkg/model/config_test.go#L236

Reply:
so i think we can make it support name.namespace

Reply:
we can do that in separated PRs, we can hold this first.